src/app/util/attribute-storage.cpp

```diff
@@ -336,6 +337,98 @@ CHIP_ERROR emberAfSetDynamicEndpoint(uint16_t index, EndpointId id, const EmberA
     emAfEndpoints[index].deviceTypeList = deviceTypeList;
     emAfEndpoints[index].endpointType   = ep;
     emAfEndpoints[index].dataVersions   = dataVersionStorage.data();
+
+    // Start the endpoint off as disabled.
+    emAfEndpoints[index].bitmask.Clear(EmberAfEndpointOptions::isEnabled);
+    emAfEndpoints[index].parentEndpointId = parentEndpointId;
+
+    emberAfSetDynamicEndpointCount(MAX_ENDPOINT_COUNT - FIXED_ENDPOINT_COUNT);
+
+    // Initialize the data versions.
+    size_t dataSize = sizeof(DataVersion) * serverClusterCount;
+    if (dataSize != 0)
+    {
+        if (Crypto::DRBG_get_bytes(reinterpret_cast<uint8_t *>(dataVersionStorage.data()), dataSize) != CHIP_NO_ERROR)
+        {
+            // Now what?  At least 0-init it.
+            memset(dataVersionStorage.data(), 0, dataSize);
+        }
+    }
+
+    // Now enable the endpoint.
+    emberAfEndpointEnableDisable(id, true);
+
+    emberMetadataStructureGeneration++;
+    return CHIP_NO_ERROR;
+}
+#else
+    return emberAfSetDynamicEndpointWithEpUniqueId(index, id, ep, dataVersionStorage, deviceTypeList, CharSpan{}, parentEndpointId);
```

Comment:
This code duplication is really sub-optimal.  I would suggest just having the internal function unconditionally defined and ifdef out the part that uses `endpointUniqueId` in it.